### PR TITLE
Csv multitank limit

### PR DESCRIPTION
**Pull request by mturkia**
### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

Ensure the cylinder amount on header line comes from a single dive. This patch takes the last dive in log to be exported as reference.
Export at most as many cylinders as the header line has and print empty strings for non-existent cylinders if one dive has less cylinders than the header.
See #2830 @p-neves commented about unreal number of cylinders in export.

### Changes made:

1) Use exported cylinder amount from a single dive
2) Print at most as many cylinders as the header has
3) Print empty cylinders if required

### Related issues:

#2830 

### Release note:

no need
### Mentions:

@p-neves

### CODE REVIEW

File: xslt/xml2manualcsv.xslt [hunk 1]
```diff
@@ -38,7 +38,7 @@
           $fs)"/>
 
         <!-- Print cylinder info according to the amount of cylinders in dive -->
-        <xsl:for-each select="dive/cylinder|trip/dive/cylinder">
+        <xsl:for-each select="dive[position()=last()]/cylinder|trip[position()=last()]/dive[position()=last()]/cylinder">
```
Comment (dirkhh) [inline xslt/xml2manualcsv.xslt:44]: so if the last dive has only one cylinder, we don't get additional cylinders from an earlier dive?

**Review by dirkhh — APPROVED** [review]
I think this is an improvement, so I'll merge - but look at my comment... I know nothing of XSLT, but is it possible to determine the max number of cylinders over all the dives being exported?

File: xslt/xml2manualcsv.xslt (hunk 1 above)
Comment (mturkia) [reply xslt/xml2manualcsv.xslt:44]: yep, that is how it goes. I think using the last dive as reference is more logical than first dive. And with XSLT it is not feasible to look for the max amount of cylinders in any of the dives to be exported (due to poor performance). So this chosen compromise appeared to make most sense to me.

**Comment by p-neves** [comment]
Seems ok with my divelog. Thanks Miika!

**Pull request merged**